src/backend/storeManagers/legendary/library.ts

```diff
@@ -584,12 +617,12 @@ function loadFile(fileName: string): boolean {
 async function loadAll(): Promise<string[]> {
   if (existsSync(legendaryMetadata)) {
     const loadedFiles: string[] = []
-    allGames.forEach((appName) => {
-      const wasLoaded = loadFile(appName + '.json')
+    for (const appName of allGames) {
+      const wasLoaded = await loadFile(appName + '.json')
```

Comment:
Can you elaborate more on this change?
Not sure if this can impact or not big libraries loading.

Reply:
ah yeah so loadFile was originally synchronous. I made it async to fetch hp metadata inside it but then decided it would be better to cache it as a map so I then reverted it back to sync. The await does nothing here. I will remove it

I switched from forEach to for of because forEach does not wait for promises to resolve but I can await the promise in the for of loop

performance here is identical to before I believe